### ISSUE TITLE
Wrapper submitted twice, probably because of the handling of HELD jobs

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#844]
In GitLab by @mcastril on Jul 6, 2022, 11:05

@dbeltrankyl, experiment a28v has been running two parallel identical wrappers since yesterday at midnight.

The wrapper could have failed. Then Autosubmit submitted a new wrapper, while it set the HELD one as QUEUING.

In `/gpfs/archive/bsc32/esarchive/autosubmit/a28v/tmp/ASLOGS/20220704_172515_run.log`

```
2022-07-06 00:18:55,454 Successful check job command: sacct -n -X --jobs 23777249 -o "State"
2022-07-06 00:18:55,454 Wrapper job a28v_ASThread_16558479437309_5040_7 changed from RUNNING to status COMPLETED
2022-07-06 00:18:55,455 Checking inner jobs status
2022-07-06 00:18:56,666 a28v_19500101_fc0_920_SIM job seems to have completed: checking...
2022-07-06 00:18:56,681 Job a28v_19500101_fc0_920_SIM is COMPLETED
2022-07-06 00:18:56,747 a28v_19500101_fc0_920_SIM_STAT file have been transfered
2022-07-06 00:19:01,716 Custom directives from platform.conf: None
2022-07-06 00:19:01,717 Custom directives from platform.conf: None
2022-07-06 00:19:03,095 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.out File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:03,095 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.err File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:03,319 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_923_SIM_STAT File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:08,323 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_923_SIM_STAT File still no exists.. waiting 10s for a new retry ( retries left: 2)
2022-07-06 00:19:09,457 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.out File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:09,457 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.err File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:18,332 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_923_SIM_STAT File still no exists.. waiting 15s for a new retry ( retries left: 1)
2022-07-06 00:19:20,807 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.out File still no exists.. waiting 5s for a new retry ( retries left: 3)
2022-07-06 00:19:20,808 /gpfs/scratch/cns16/cns16454/a28v/LOG_a28v/a28v_19500101_fc0_920_SIM.cmd.err File still no exists.. waiting 5s for a new retry ( retries left: 3)
```

```
2022-07-06 00:22:20,231 Successful check job command: sacct -n -X --jobs 23777257 -o "State"
2022-07-06 00:22:20,231 Checking inner jobs status
2022-07-06 00:22:21,352 Job a28v_ASThread_16569425283418_5040_7 is HELD
2022-07-06 00:22:21,352 End of checking
2022-07-06 00:22:21,354 Updating FAILED jobs
2022-07-06 00:22:21,363 Resetting job: a28v_19500101_fc0_921_SIM status to: READY for retrial...
2022-07-06 00:22:21,363 Resetting job: a28v_19500101_fc0_922_SIM status to: WAITING for parents completion...
2022-07-06 00:22:21,363 Resetting job: a28v_19500101_fc0_923_SIM status to: WAITING for parents completion...
2022-07-06 00:22:21,364 Updating WAITING jobs
2022-07-06 00:22:21,366 Updating WAITING jobs eligible for be prepared
2022-07-06 00:22:21,368 Updating Held jobs
2022-07-06 00:22:21,369 Setting job: a28v_ASThread_16569425283418_5040_7 status to: Queuing (all parents completed)...
2022-07-06 00:22:21,369 Update finished
```

```
2022-07-06 00:22:42,652 Job a28v_ASThread_16569425283418_5040_7 is QUEUING (JobHeldUser)
2022-07-06 00:22:42,652 Checking Wrapper 23802377
2022-07-06 00:22:45,825 Successful check job command: sacct -n -X --jobs 23802377 -o "State"
2022-07-06 00:22:45,826 Wrapper job a28v_ASThread_16570597418610_5040_3 changed from SUBMITTED to status QUEUING
2022-07-06 00:22:45,826 Checking inner jobs status
2022-07-06 00:22:46,946 Job a28v_ASThread_16570597418610_5040_3 is QUEUING (Priority)
2022-07-06 00:22:46,947 End of checking
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 6, 2022, 11:05

@emchamarro @eferre1 @portegam

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 19, 2022, 14:04

Since presubmission is not useful anymore I'll put it to stop, and maybe it is better to put it for AS 4+

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 27, 2023, 09:41

I think this is related to the submission bug encountered last month and should have been fixed already in 3.14 and 3.15
